### PR TITLE
Add the Green-Naghdi objective rate to the new mechanics system

### LINKED ISSUE #21166: Add Green-Naghdi objective rate
## Reason

For what it's worth, the Green-Naghdi rate is rotational objective, which might behave more like the existing integration method in tensor mechanics. So adding it will be bring the gap between the new and the old systems closer.

## Design

Add the Green-Naghdi objective rate.

## Impact

This is a new way to integrate the stress.

### PR DESCRIPTION
**Pull request by hugary1995**
close #21166

### CODE REVIEW

**Comment by hugary1995** [comment]
Since I reorganized the test directory in #20744, this will create conflicts with that PR. I'm fine with first merging either one. I'll just have to resolve the conflicts after one is merged.

**Comment by hugary1995** [comment]
This PR need the factorized R2T PR #20376 which is stacked on the Mandel notation PR #18998

**Comment by reverendbedford** [comment]
I will review, but I'll hold off until #20376 is merged.

File: modules/tensor_mechanics/src/materials/lagrangian/ComputeLagrangianObjectiveStress.C
Comment (reverendbedford) [inline modules/tensor_mechanics/src/materials/lagrangian/ComputeLagrangianObjectiveStress.C:144]: I'm getting lost on what depends on what, but can't we use @dschwen new einsum stuff to do these?

File: modules/tensor_mechanics/src/materials/lagrangian/ComputeLagrangianObjectiveStress.C
Comment (reverendbedford) [inline modules/tensor_mechanics/src/materials/lagrangian/ComputeLagrangianObjectiveStress.C:165]: Same comment with einsum, probably will show up lots of places.

File: modules/tensor_mechanics/src/materials/lagrangian/ComputeHypoelasticStVenantKirchhoffStress.C
Comment (reverendbedford) [inline modules/tensor_mechanics/src/materials/lagrangian/ComputeHypoelasticStVenantKirchhoffStress.C:56]: Use the new `einsum` stuff?

File: modules/tensor_mechanics/src/materials/lagrangian/ComputeHypoelasticStVenantKirchhoffStress.C
Comment (reverendbedford) [inline modules/tensor_mechanics/src/materials/lagrangian/ComputeHypoelasticStVenantKirchhoffStress.C:66]: Uhh, do we want to start introducing this now or do it all at once later on?

**Review by reverendbedford — CHANGES_REQUESTED** [review]
I'm getting lost on the order of what needs to go in first, but if we get what I'm going to defiantly call the `einsum` PR in before this that would simplify all the mixed products.  Also the docs need to be cleaned up to reflect the final state of what was implemented.

**Comment by reverendbedford** [comment]
> 

Nevermind, I just tried to only comment on the changes that aren't in another PR.

File: modules/tensor_mechanics/src/materials/lagrangian/ComputeHypoelasticStVenantKirchhoffStress.C
Comment (hugary1995) [reply modules/tensor_mechanics/src/materials/lagrangian/ComputeHypoelasticStVenantKirchhoffStress.C:67]: Let me do the right thing here. I'll correct everything else in a separate PR.

File: framework/include/utils/RankTwoTensor.h
Comment (dschwen) [inline framework/include/utils/RankTwoTensor.h:81]: Are you even using these? I see `MathUtils::transpose()` being used below.

File: framework/src/utils/RankTwoTensor.C
Comment (dschwen) [inline framework/src/utils/RankTwoTensor.C:44]: Yeah, I don't think those should be added. And if they were needed, why aren't they static?

Comment (hugary1995) [reply framework/src/utils/RankTwoTensor.C:44]: I don't recall adding these methods myself. Maybe they are the result of an careless merge.

File: framework/include/utils/RankTwoTensor.h
Comment (hugary1995) [reply framework/include/utils/RankTwoTensor.h]: Removed.

File: framework/include/utils/RankFourTensor.h
Comment (hugary1995) [reply framework/include/utils/RankFourTensor.h:375]: What I learnt from some recent libtorch code reading is that in theory we could implement an einsum using std::string_view, and if we are careful enough the method dispatch should happen at compile time.

Comment (hugary1995) [reply framework/include/utils/RankFourTensor.h:375]: I documented this in issue https://github.com/idaholab/moose/issues/21568. So marking this as resolved, for now.

**Review by dschwen — DISMISSED** [review]

**Comment by dschwen** [comment]
Double check if those conflicts are resolved correctly

**Comment by hugary1995** [comment]
The merge commit looks good to me. The conflict arises from me trying to consolidate input files for large/small kinematics. So if all regression tests pass, I think this is good to go.

**Review by dschwen — APPROVED** [review]

**Pull request merged**
